cmd/servicelog/post.go

```diff
@@ -24,175 +26,208 @@ import (
+		log.Error("program abruptly terminated, performing clean-up...")
+		o.cleanUp(clusters)
+		log.Fatal("servicelog post command terminated")
+	}()
 
-		// If this is a dry-run, don't proceed further.
-		if isDryRun {
-			return
+	for _, cluster := range clusters {
+		request, err := o.createPostRequest(ocmClient, cluster)
+		if err != nil {
+			o.failedClusters[cluster.ExternalID()] = err.Error()
+			continue
 		}
 
-		if !skipPrompts {
-			err = ctlutil.ConfirmSend()
-			if err != nil {
-				log.Fatal(err)
-			}
+		response, err := sendRequest(request)
+		if err != nil {
+			o.failedClusters[cluster.ExternalID()] = err.Error()
+			continue
 		}
 
-		// Handler if the program terminates abruptly
-		go func() {
-			sigchan := make(chan os.Signal, 1)
-			signal.Notify(sigchan, os.Interrupt)
-			<-sigchan
-
-			// perform final cleanup actions
-			log.Error("program abruptly terminated, performing clean-up...")
-			cleanUp(clusters)
-			log.Fatal("servicelog post command terminated")
-		}()
-
-		for _, cluster := range clusters {
-			request, err := createPostRequest(ocmClient, cluster)
-			if err != nil {
-				failedClusters[cluster.ExternalID()] = err.Error()
-				continue
-			}
-
-			response, err := sendRequest(request)
-			if err != nil {
-				failedClusters[cluster.ExternalID()] = err.Error()
-				continue
-			}
-
-			check(response, Message)
-		}
+		o.check(response, o.Message)
+	}
 
-		printPostOutput()
-	},
+	o.printPostOutput()
+	return nil
 }
 
-func init() {
-	// define required flags
-	postCmd.Flags().StringVarP(&template, "template", "t", defaultTemplate, "Message template file or URL")
-	postCmd.Flags().StringArrayVarP(&templateParams, "param", "p", templateParams, "Specify a key-value pair (eg. -p FOO=BAR) to set/override a parameter value in the template.")
-	postCmd.Flags().BoolVarP(&isDryRun, "dry-run", "d", false, "Dry-run - print the service log about to be sent but don't send it.")
-	postCmd.Flags().StringArrayVarP(&filterParams, "query", "q", filterParams, "Specify a search query (eg. -q \"name like foo\") for a bulk-post to matching clusters.")
-	postCmd.Flags().BoolVarP(&skipPrompts, "yes", "y", false, "Skips all prompts.")
-	postCmd.Flags().StringArrayVarP(&filterFiles, "query-file", "f", filterFiles, "File containing search queries to apply. All lines in the file will be concatenated into a single query. If this flag is called multiple times, every file's search query will be combined with logical AND.")
-	postCmd.Flags().StringVarP(&clustersFile, "clusters-file", "c", clustersFile, `Read a list of clusters to post the servicelog to. the format of the file is: {"clusters":["$CLUSTERID"]}`)
-	postCmd.Flags().BoolVarP(&internalOnly, "internal", "i", false, "Internal only service log. Use MESSAGE for template parameter (eg. -p MESSAGE='My super secret message').")
+func (o *PostCmdOptions) check(response *sdk.Response, clusterMessage servicelog.Message) {
+	body := response.Bytes()
+	if response.Status() < 400 {
```

Comment:
**thought:** A bit unrelated to this PR, but it seems odd to consider anything with code < 400 as "good response".

Reply:
Yeah it's definitely a weird way to confirm a "good response". That being said, I don't want to touch anything unnecessary in this command given how important its use is 😅